### PR TITLE
.NET Core, Console & Npm Runners

### PR DESCRIPTION
**Pull request by Drewan**

### CODE REVIEW

File: LightBlue.MultiHost/Runners/RoleRunner.cs [hunk 1]
```diff
@@ -46,6 +46,30 @@ public RoleRunner(Role role)
 
         public void Start()
         {
+            if (_role.RoleName == "DotNetCore")
```
Comment (eric-winkler) [inline LightBlue.MultiHost/Runners/RoleRunner.cs:49]: can we apply a bit of DRY here, perhaps something like;
```
var role =
    _role.RoleName == "DotNetCore" ? RunnerFactory.CreateDotNetCoreRunner(_role) :
    _role.RoleName == "Console" ? RunnerFactory.CreateConsoleRunner(_role) : 
    ...

_resources.Add(role);
role.Start();
```

Comment (eric-winkler) [reply LightBlue.MultiHost/Runners/RoleRunner.cs:49]: In fact, perhaps this decision logic is actually an internal concern of the `RunnerFactory`? ie
`var role = RunnerFactory.CreateRunner(_role)` ?

File: LightBlue.MultiHost/Runners/NpmRunner.cs [hunk 2]
```diff
@@ -0,0 +1,90 @@
+﻿using System;
```
Comment (eric-winkler) [inline LightBlue.MultiHost/Runners/NpmRunner.cs:1]: It's interesting that `NpmRunner`, `DotNetCoreRunner` and `ConsoleRunner` only differ by a few lines.  Feels like the actual runner is a `ShellRunner` or `ExternalProcess` runner, and the npm\dotnetcore\console aspects are just injected path & args (to internally pass down to ProcessStartInfo)

Comment (Drewan) [reply LightBlue.MultiHost/Runners/NpmRunner.cs:1]: Currently the axis of change is reflected in the name of the runner (.NET core, NPM, Console).  To me this seems like a good application of SRP that would support changes along those axis.  The ShellRunner would couple those three responsibilities in favour of DRY, is that better?

File: LightBlue.MultiHost/Runners/RoleRunner.cs (hunk 1 above)
Comment (Drewan) [reply LightBlue.MultiHost/Runners/RoleRunner.cs:49]: Agree encapsulating that logic inside the factory is a better design but I wasn't really looking to refactor just following what's already in place (CreateForWebsite, CreateForWorkerRole etc).

**Review by eric-winkler — APPROVED** [review]
This is some neat new functionality.

Although I'm not convinced that the abstraction is correct (ie I think docker support will be added by just changing the command line again) I'm content that there's been proper consideration given to how this can be extended.

**Pull request merged**